javascript/sdk/lib/controllers/build.ts

```diff
@@ -10,17 +10,39 @@ export default class BuildController {
 
   /** Create or update a test */
   async createTest(
-    testId: string,
     name: string,
     unit: string,
     advisory?: string,
+    testId?: string,
     options: RequestOptions = {}
-  ): Promise<void> {
-    await this.#client.requestWithAuth(`/build/tests/${testId}`, {
+  ): Promise<Test> {
+    const response = await this.#client.requestWithAuth(`/build/tests`, {
       method: "POST",
-      body: JSON.stringify({ name, unit, advisory }),
+      body: JSON.stringify({ name, unit, advisory, testId }),
       ...options,
     });
+
+    return await response.json();
+  }
+
+  /** Update a test */
+  async updateTest(
+    testId: string,
+    name?: string,
+    unit?: string,
+    advisory?: string,
+    options: RequestOptions = {}
+  ): Promise<Test> {
+    const response = await this.#client.requestWithAuth(
+      `/build/tests/${testId}`,
+      {
+        method: "POST",
+        body: JSON.stringify({ name, unit, advisory }),
```

Comment:
so if user doesn't provide an advisory, does the body not contain the "advisory" key at all?

Reply:
yep, is that not correct?

Reply:
it will be passed in as undefined I believe? @kenrick can you verify that statement

Reply:
the updates work as "if the key is in the body, we'll update it's field. And if the key is not in the body, we leave the field as it is"

So we don't want keys to be passed through with default values, b/c if the user isn't explicitly setting the input param, they probably mean to leave it unchanged

Reply:
so same for updateEndpoint - we don't want to default host/edr_id/tags to "" or all those will actually be updated to empty string in the BE

Reply:
```js
JSON.stringify({ advisory: undefined })
```
will produce

```
"{}"
```

Undefined it not a valid JSON value. So the property would be omitted.